src/vivarium/framework/resource/manager.py

```diff
@@ -137,16 +121,17 @@ def _get_resource_group(
         --------
         :class:`ResourceGroup`
         """
+        resources_ = [Column(r) if isinstance(r, str) else r for r in resources]
         dependencies_ = [Column(d) if isinstance(d, str) else d for d in dependencies]
-        if not resource_names:
+
+        if not resources_:
             # We have a "producer" that doesn't produce anything, but
```

Comment:
What's an example of a null producer?

Reply:
A NullProducer happens when a component has an `on_initialize_simulants` method, but that method doesn't create any columns. This could happen if the component wanted to initialize some attribute of itself using columns or pipelines created by other components.